File: DSA_Codesheet/Python/triplet_sum.py
def find_triplet_sum(arr, n, X):
    # Sort the input array
    arr.sort()

    for i in range(n - 2):
        left = i + 1
        right = n - 1

        while left < right:
            current_sum = arr[i] + arr[left] + arr[right]

            if current_sum == X:
                return 1  # Triplet found

            if current_sum < X:
                left += 1
            else:
                right -= 1

    return 0  # No triplet found

# Input
n, X = map(int, input().split())
arr = list(map(int, input().split()))

# Check if a triplet with the given sum X exists in the array
result = find_triplet_sum(arr, n, X)
print(result)

File: DSA_Codesheet/Java/Triplet_Sum.java
import java.util.Arrays;

public class Triplet_Sum {
    public static void main(String[] args) {
        int[] arr = {1,2,4,3,6};
        System.out.println(threeSum(arr,10));

    }
    static boolean threeSum(int[] arr,int x){
        Arrays.sort(arr);
        int n = arr.length;
        for (int i = 0; i < n; i++) {
            int j = i+1;
            int k = arr.length-1;
            while (j<k){
                int sum = arr[i]+arr[j]+arr[k];
                if(sum==x){
                    return true;
                } else if (sum<x) {
                    j++;
                }else{
                    k--;
                }
            }
        }
        return false;
    }
}
